lib/internal/Magento/Framework/Reflection/DataObjectProcessor.php

```diff
@@ -102,9 +102,9 @@ public function buildOutputDataArray($dataObject, $dataObjectType)
                     continue;
                 }
             } else {
-                if (is_object($value) && !($value instanceof Phrase)) {
+                if (is_object($value) && !is_iterable($value) && !($value instanceof Phrase)) {
```

Comment:
Better to check on Marker interface here. I can have the class that is iterable but does not the collection

Reply:
@kandy to you mean `\Magento\Framework\Data\CollectionDataSourceInterface`?